File: model/src/main/java/net/artux/pda/model/map/SpawnModel.java
package net.artux.pda.model.map;

import net.artux.pda.model.user.Gang;

import java.io.Serializable;
import java.util.HashMap;
import java.util.List;
import java.util.Set;

import lombok.Data;

@Data
public class SpawnModel implements Serializable {

    private int id;
    private String title;
    private String description;
    private Gang group;
    private Strength strength;
    private int r;
    private int n;
    private String pos;
    private HashMap<String, String> data;
    private HashMap<String, List<String>> actions;
    private HashMap<String, List<String>> condition;

    public Set<String> getParams() {
        if (data == null)
            data = new HashMap<>();
        return data.keySet();
    }

}

File: model/src/main/java/net/artux/pda/model/map/SpawnModel.kt
package net.artux.pda.model.map

import net.artux.pda.model.user.Gang
import java.io.Serializable
import java.util.Collections

data class SpawnModel(
    var id: Int = 0,
    var title: String? = null,
    var description: String? = null,
    var group: Gang? = null,
    var strength: Strength? = null,
    var r: Int = 0,
    var n: Int = 0,
    var pos: String = "500:500"
) : Serializable {

    var data: Map<String, List<String>> = mutableMapOf()
    var actions: Map<String, List<String>> = mutableMapOf()
    var condition: Map<String, List<String>> = mutableMapOf()

    fun getParams(): Set<String> {
        return data.keys
    }
}